Home work

yuraej opened the pull request.

Commit:
added two classes are "...SettingsModal" & "...UploadFilesModal"
task is not done

dzmitryhanets committed:
added new methods

Commit:
task is done
added: class"TestData", file"volk.jpg'

dzmitryrak commented on src/test/java/pages/IntegriVideoChatPage.java:
```diff
@@ -1,66 +1,93 @@
+
+    public IntegriVideoChatPage(WebDriver driver){
+        this.driver = driver;
+        wait = new WebDriverWait(driver,5);
```
после запятой пробел

dzmitryrak commented on src/test/java/pages/IntegriVideoChatPage.java:
```diff
@@ -1,66 +1,93 @@
+        wait = new WebDriverWait(driver,5);
+    }
+
+    public void openPage(){
```
всегда перед фигурной скобкой - пробел. Используй автоформатирование

dzmitryrak commented on src/test/java/pages/IntegriVideoChatPage.java:
```diff
@@ -1,66 +1,93 @@
+        wait.until(ExpectedConditions.textToBe(textMassage, text));
+       return driver.findElement(textMassage).getText();
+    }
+    public String  checkLinkByMassage(String link){
```
между методами должна быть пустая строка

dzmitryrak commented on src/test/java/pages/IntegriVideoChatPage.java:
```diff
@@ -1,66 +1,93 @@
+    By textMassage = By.className("integri-chat-message-text");
+    By edit = By.cssSelector("span.iv-icon.iv-icon-pencil.integri-chat-edit-message");
+    By editText = By.tagName("textarea");
+    By delete = By.cssSelector("span.iv-icon.iv-icon-trash2.integri-chat-remove-message");
```
очень много указываешь лишних классов при поиске. Хватает одного обычно, который реально говорит о том что это за элемент

yuraej replied:
это не лишние классы, без названия после второй точки элемент не определяется.

yuraej replied:
точкой заменяется пробел

yuraej replied:
в названии класса: "iv-icon iv-icon-pencil integri-chat-edit-message"

dzmitryrak commented on src/test/java/pages/IntegriVideoUploadFilesModal.java:
```diff
@@ -0,0 +1,25 @@
+        wait = new WebDriverWait(driver, 7);
+    }
+
+    public void setDragAndDrop (String linkToFile){
```
почему ты так назвал этот метод? Это ведь метод uploadFile()

yuraej replied:
потому что элемент так называеся, при нажатии на который я могу загружать файлы, а 
элемента с названием upload File на странице нет

dzmitryrak commented on src/test/java/tests/CheckIntegriVideo.java:
```diff
@@ -2,57 +2,156 @@
+        assertEquals(chat.checkMessage("test2"), "test2");
+    }
+
+    @Test (priority = 3)
```
зачем приоритеты?

dzmitryrak commented on src/test/java/tests/CheckIntegriVideo.java:
```diff
@@ -2,57 +2,156 @@
+    @Test (priority = 4)
+    public void checkDeleteMessage() {
+        IntegriVideoChatPage chat = new IntegriVideoChatPage(driver);
+        chat.deleteMessage();
```
тесты не должны быть зависимыми один от одного. Поэтому в тесте про удаление - сначала шлешь - потом удаляешь

dzmitryrak commented on src/test/java/tests/CheckIntegriVideo.java:
```diff
@@ -2,57 +2,156 @@
+    public void checkCopyCode() throws IOException, UnsupportedFlavorException {
+        IntegriVideoChatPage chat = new IntegriVideoChatPage(driver);
+        String textCode = chat.checkCode();
+        String textCodeMemory = (String) Toolkit.getDefaultToolkit().getSystemClipboard().getData(DataFlavor.stringFlavor);
```
я говорил этот метод запихнуть внутрь PageObject

dzmitryrak commented on src/test/java/tests/CheckIntegriVideo.java:
```diff
@@ -2,57 +2,156 @@
+    public void InputSettingsModal(){
+        IntegriVideoSettingsModal settingsModal = new IntegriVideoSettingsModal(driver);
+        settingsModal.inputSettings();
+        settingsModal.inputUserName("BONE");
```
в переменную вынести слово

dzmitryrak commented on src/test/java/tests/CheckIntegriVideo.java:
```diff
@@ -2,57 +2,156 @@
+    }
+
+    @Test (priority = 10)
+    public void RenameSettingsModal(){
```
имена методов с маленькой буквы

Commit:
error correction

Commit:
renamed setDragAndDrop method to upLoadFiles

Commit:
added new classes:
- LoginPageFactory
- LoginPageObject
- EnterByLoginPageFactory
- EnterByLoginPageObject

added new data in class TestData - "urlLoginPage"

dzmitryrak commented on src/test/java/pages/IntegriVideoUploadFilesModal.java:
```diff
@@ -0,0 +1,26 @@
+        driver.findElement(dragAndDrop).click();
+        driver.findElement(browse).sendKeys(linkToFile);
+        driver.findElement(dragAndDropStart).click();
+        wait.until(ExpectedConditions.elementToBeClickable(dragAndDropStart));
```
сначала ждем, потом кликаем

dzmitryrak commented on src/test/java/tests/CheckIntegriVideo.java:
```diff
@@ -2,57 +2,166 @@
+        chat.sendMassageByEnter();
+        assertEquals("<html><body><p>test</p></body></html>", chat.checkMessage("<html><body><p>test</p></body></html>"));
         chat.settingsInput();
         chat.dragAndDrop();
```
бесмысленные проверки с 
        chat.settingsInput();
        chat.dragAndDrop();

удали

dzmitryrak commented on src/test/java/pages/IntegriVideoChatPage.java:
```diff
@@ -1,66 +1,101 @@
+        driver.findElement(inviteLink).click();
+    }
+
+    public void inputMessage(String massage) {
```
не massage, а message

dzmitryrak commented on src/test/java/pages/IntegriVideoUploadFilesModal.java:
```diff
@@ -0,0 +1,26 @@
+
+    public IntegriVideoUploadFilesModal(WebDriver driver) {
+        this.driver = driver;
+        wait = new WebDriverWait(driver, 7);
```
7 милисекнд что ли? мало
